generate.py

```diff
@@ -102,7 +105,7 @@ def generate(args):
     ext = args.vehicle_types.split('.')[-1]
     if ext != "json":
         print("File extension mismatch: vehicle type file should be .json")
-    with open(args.vehicle_types) as f:
+    with open(path.join(target_path, args.vehicle_types)) as f:
```

Comment:
The vehicle_types file is only used for generating the file, NOT in the simulation. The "all paths are relative to the target path" is referring to the paths in the scenario.json, not all paths during generation as well. The vehicle_types path is given as absolute or relative to the working directory - irrespective of the target_path.
Having the files needed by the simulation as relative to the scenario file location just makes it easier to find them again during the simulation (which might have a different working directory), without having to resort to absolute paths. The path to the vehicle_types file is not saved in the simulation file, so there is no need to adapt the path.
For clarity's sake, you could change the comment in line 80 to `file paths in the output file are relative to that file`.